src/comps/Header.tsx

```diff
@@ -1,36 +1,150 @@
 import Image from "next/image";
 import Link from "next/link";
+import { AppConfig, UserSession } from "@stacks/connect";
+import { useEffect, useState } from "react";
+
+const appConfig = new AppConfig(["store_write"]);
+const userSession = new UserSession({ appConfig });
+import { AnimatePresence } from "framer-motion";
+import { useAtom } from "jotai";
+import {
+  isConnectedAtom,
+  showConnectWalletAtom,
+  userDataAtom,
+} from "@/util/atoms";
+
+import ConnectWallet from "./ConnectWallet";
+import { useWallet } from "@/util/WalletContext";
 
 const Header = () => {
+  const [userData, setUserData] = useAtom(userDataAtom);
+
+  const { handleSignOut } = useWallet();
+
+  const [showConnectWallet, setShowConnectWallet] = useAtom(
+    showConnectWalletAtom
+  );
+
+  const [isConnected, setIsConnected] = useAtom(isConnectedAtom);
+
+  // useEffect ot check if user is signed in
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      const userData = userSession.loadUserData();
+      setUserData(userData);
+      setIsConnected(true);
+    }
+  }, []);
+
+  const renderUserWalletInf = () => {
+    // show the first 4 and last 4 of their stx wallet and their segiwt address
+    if (userData === null) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
+
+    // ensure the profile is not undefined
+    if (userData?.profile === undefined) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
+
+    // ensure p2wpkh is not undefined
+    console.log(userData);
+    if (userData?.profile.btcAddress === undefined) {
+      return (
+        <button
+          onClick={() => handleSignOut()}
+          className="   px-4 py-2 rounded-md border-2 border-orange"
+        >
+          <h3 className="font-Matter text-xs text-orange font-semibold	 tracking-wide">
+            DISCONNECT WALLET
+          </h3>
+        </button>
+      );
+    }
```

Comment:
Those three cases returns the same thing. If that's expected, can we merge them in a single condition? 

Also it seems the only difference is if we show the two `h4` or not. If so can we not do the early return to avoid having a duplicated `button` that we need to remember to update in both places?

Reply:
For the time being these we're removed, this requires further logic based on which wallet is loaded and what is available